packages/browser-wallet-api-helpers/src/wallet-api-types.ts

```diff
@@ -4,15 +4,26 @@ import type {
     AccountTransactionType,
 } from '@concordium/web-sdk';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type WalletEventHandler<T = any> = (payload: T) => void;
+/**
+ * An enumeration of the events that can be emitted by the WalletApi.
+ */
+export enum EventType {
+    AccountChanged = 'accountChanged',
+    ChainChanged = 'chainChanged',
+}
 
-export interface WalletApi {
-    /**
-     * React to changes to the selected account in the Concordium Wallet. Note that to get the initially selected account on load, the "connect" method should be used.
-     * @param handler a handler function called with the account address of the selected account.
-     */
-    addChangeAccountListener(handler: WalletEventHandler<string>): void;
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+interface Listeners<T extends EventType, EventListener extends (...args: any[]) => void> {
+    on(eventName: T | `${T}`, listener: EventListener): this;
+    once(eventName: T | `${T}`, listener: EventListener): this;
+    addListener(eventName: T | `${T}`, listener: EventListener): this;
+    removeListener(eventName: T | `${T}`, listener: EventListener): this;
+}
+
+type EventListeners = Listeners<EventType.AccountChanged, (accountAddress: string) => void> &
+    Listeners<EventType.ChainChanged, (chain: string) => void>;
```

Comment:
```suggestion
type EventListener<Args extends any[]> =  (...args: Args) => void;

// eslint-disable-next-line  @typescript-eslint/no-explicit-any
interface Listeners<T extends EventType, Args extends any[]> {
    on(eventName: T | `${T}`, listener: EventListener<Args>): this;
    once(eventName: T | `${T}`, listener: EventListener<Args>): this;
    addListener(eventName: T | `${T}`, listener: EventListener<Args>): this;
    removeListener(eventName: T | `${T}`, listener: EventListener<Args>): this;
}

type EventListeners = Listeners<EventType.AccountChanged, [accountAddress: string]> &
    Listeners<EventType.ChainChanged, [chain: string]>;
```

To avoid specifying the entire function for each event. (Seems a bit overkill currently though :thinking: )

Reply:
It's very nice and it just works, not overkill at all.